core/src/main/java/ninja/soroosh/chatopia/core/connectors/telegram/ResponseHandler.java

```diff
@@ -40,59 +48,91 @@ public void init() {
+        } else if (response instanceof PhotoResponse || response instanceof VideoResponse) {
+            result = MediaResponseTemplate(response);
+        }
+        return result;
+    }
+
+    private Object MediaResponseTemplate(Response response) {
+        Object result = null;
+        if (response instanceof PhotoResponse photoResponse) {
             if (photoResponse.getId() != null) {
-                final var requestBody = new TelegramSendPhoto(chatId, photoResponse.getId(), photoResponse.getCaption());
+                final var requestBody = new TelegramSendMedia(chatId, photoResponse.getId(), photoResponse.getCaption());
                 result = restTemplate.postForEntity(
-                        String.format("https://api.telegram.org/bot%s/sendPhoto", key), requestBody
+                        BASEURL.concat(key).concat("/").concat(SEND_PHOTO), requestBody
                         , Object.class);
             } else if (photoResponse.getUrl() != null) {
-                final var requestBody = new TelegramSendPhoto(chatId, photoResponse.getUrl().toString(), photoResponse.getCaption());
+                final var requestBody = new TelegramSendMedia(chatId, photoResponse.getUrl().toString(), photoResponse.getCaption());
                 result = restTemplate.postForEntity(
-                        String.format("https://api.telegram.org/bot%s/sendPhoto", key), requestBody
+                        BASEURL.concat(key).concat("/").concat(SEND_PHOTO), requestBody
                         , Object.class);
             } else {
-                // TODO: extract every response type to its handler
-                HttpHeaders headers = new HttpHeaders();
-                headers.setContentType(MediaType.MULTIPART_FORM_DATA);
-
-                MultiValueMap<String, String> fileMap = new LinkedMultiValueMap<>();
-                ContentDisposition contentDisposition = ContentDisposition
-                        .builder("form-data")
-                        .name("photo")
-                        .filename("photo")
-                        .build();
-                fileMap.add(HttpHeaders.CONTENT_DISPOSITION, contentDisposition.toString());
-
-                HttpEntity<byte[]> fileEntity = null;
-                try {
-                    fileEntity = new HttpEntity<>(photoResponse.getPhotoStream().readAllBytes(), fileMap);
-                } catch (IOException e) {
-                    // TODO: Handle error
-                    e.printStackTrace();
-                }
-
-                MultiValueMap<String, Object> form = new LinkedMultiValueMap<>();
-                form.add("photo", fileEntity);
-                form.add("chat_id", chatId);
-                form.add("caption", photoResponse.getCaption());
-                HttpEntity<MultiValueMap<String, Object>> requestEntity
-                        = new HttpEntity<>(form, headers);
-
+                result = formGenerator(response, TYPE_PHOTO);
```

Comment:
extract the API call out from the formGenerator. 
let it to just prepare the request.